cas-compute/src/symbolic/expr/mod.rs

```diff
@@ -441,50 +441,52 @@ impl PartialEq for Expr {
     }
 }
 
-/// Implements the [`std::hash::Hash`] trait for the [`Expr`] enum, which represents
+/// Implements the [`Hash`] trait for the [`Expr`] enum, which represents
 /// different types of mathematical expressions. This implementation is necessary
-/// because [`PartialEq`] is also manually implemented, and the automatically
-/// derived [`Hash`] implementation would not guarantee consistency with [`PartialEq`].
-impl std::hash::Hash for Expr {
-    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
-        // Hash the discriminant to differentiate between enum variants.
+/// because [`PartialEq`] is also manually implemented. The derived implementation
+/// of [`Hash`] would not guarantee consistency with [`PartialEq`], especially for 
+/// commutative operations like addition and multiplication.
+impl Hash for Expr {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        // Hash the discriminant to differentiate between enum variants
         let self_discr = std::mem::discriminant(self);
-        std::hash::Hash::hash(&self_discr, state);
+        Hash::hash(&self_discr, state);
```

Comment:
```suggestion
        self_discr.hash(state);
```

Use `self_discr.hash(state)` to be consistent with other usages of `Hash` below this line.